Kernel/ProcessSpecificExposed.cpp

```diff
@@ -242,6 +242,42 @@ RefPtr<ProcFSExposedComponent> ProcFSProcessFileDescriptions::lookup(StringView
     return procfd_fd;
 }
 
+class ProcFSProcessPledge final : public ProcFSProcessInformation {
+public:
+    static NonnullRefPtr<ProcFSProcessPledge> create(const ProcFSProcessFolder& parent_folder)
+    {
+        return adopt_ref(*new (nothrow) ProcFSProcessPledge(parent_folder));
+    }
+
+private:
+    explicit ProcFSProcessPledge(const ProcFSProcessFolder& parent_folder)
+        : ProcFSProcessInformation("pledge"sv, parent_folder)
+    {
+    }
+    virtual bool output(KBufferBuilder& builder) override
+    {
+        auto parent_folder = m_parent_folder.strong_ref();
+        if (parent_folder.is_null())
+            return false;
+        auto process = parent_folder->m_associated_process;
+        JsonObjectSerializer obj { builder };
+#define __ENUMERATE_PLEDGE_PROMISE(x)       \
```

Comment:
uh, is this macro necessary?

Reply:
A macro like this one is also used in Kernel/Syscalls/pledge.cpp